metadoc-js/src/main/scala/metadoc/Navigation.scala

```diff
@@ -48,13 +48,17 @@ object Navigation {
       )
   }
 
-  def currentState(
-      state: Option[State] = Option(
-        dom.window.history.state.asInstanceOf[Navigation.State]
-      ),
-      locationHash: => String = dom.window.location.hash.dropWhile(_ == '#')
-  ): Option[Navigation.State] =
-    state.orElse(parseState(locationHash))
+  def currentState(maybeState: js.Any): Option[Navigation.State] = {
```

Comment:
The name of this method could be better, maybe something like `Navigation.fromHistoryState()`